### PR TITLE
0.62.0 and 0.63.0 release notes

### PR DESCRIPTION
**Pull request by fsamuel-bs**
**Goals (and why)**: 0.62.0 release notes

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2586)

### CODE REVIEW

**Comment by fsamuel-bs** [comment]
I still owe a final read through the release notes to clean up their messaging, but formatting should be done.

File: docs/source/_static/release-notes.css [hunk 1]
```diff
@@ -22,6 +22,14 @@
   color: white;
 }
 
+.changetype.changetype-logs {
+  background: #D3D62C;
+}
+
+.changetype.changetype-metrics {
+  background: #D3D62C;
```
Comment (jeremyk-91) [inline docs/source/_static/release-notes.css]: nit: I'd suggest using the Blueprint colours - http://blueprintjs.com/docs/#core/colors, perhaps `gold4` (`F2B824`) for consistency with some of the other change-types.

File: docs/source/release_notes/release-notes.rst [hunk 2]
```diff
@@ -44,12 +50,14 @@ develop
     *    - Type
          - Change
          
-    *    - |improved|
+    *    - |metrics|
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: `|metrics|` and `|logs|` aren't quite at the same level of abstraction as `|new|`, `|fixed|` etc. Consider that if we used the internal structured changelog framework, then metrics and logs would be tags. It's OK that we use changetypes for this, for lack of a better alternative, but the type of the change should be maintained. So I'd write this change `|improved| |metrics|`.

**Comment by fsamuel-bs** [comment]
Should be ready for review. Pinging @jeremyk-91.

File: docs/source/release_notes/release-notes.rst [hunk 3]
```diff
@@ -98,132 +106,157 @@ develop
...
            The existing ``create`` methods are deprecated and will be removed by November 15th, 2017.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/2459>`__)
 
-    *    - |fixed|
-         - Async Initialization now works with TimeLock Server.
-           Previously, for Cassandra we would attempt to immediately migrate the timestamp bound from Cassandra to TimeLock on startup, which would fail if either of them was unavailable.
-           For DBKVS or other key-value services, we would attempt to ping TimeLock on startup, which would fail if TimeLock was unavailable (though the KVS need not be available).
-           (`Pull Request <https://github.com/palantir/atlasdb/pull/2518>`__)
+    *    - |devbreak|
+         - ``TransactionManagers.builder()`` no longer has a ``callingClass(..)`` method and now requires the consumer to directly specify their user agent via the previously optional method ``userAgent(..)``. All of the ``TransactionManagers.create(..)`` methods are still deprecated.
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: nit: this one doesn't seem like an improvement.

I'm not sure where to put this, though. Perhaps we could reword this to say that this is part of efforts to ensure that users explicitly provide user agents in a consistent, streamlined way.

File: docs/source/release_notes/release-notes.rst [hunk 4]
```diff
@@ -98,132 +106,157 @@ develop
...
+
+    *    - |improved|
+         - Exposes another version of ``getRanges`` that uses a configurable concurrency level when not explicitly
+           provided a value. This defaults to 8 and can be configured with the ``KeyValueServiceConfig#defaultGetRangesConcurrency`` parameter.
+           Check the full configuration docs `here <https://palantir.github.io/atlasdb/html/configuration/key_value_service_configs/index.html>`__.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2484>`__)
+
+    *    - |devbreak|
+         - Simplify and annotate the constructors for ``SerializableTransactionManager``. This should make the code of that class more maintainable.
+           If you used one of the deleted or deprecated constructors, use the static ``create`` method.
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: nit: This seems like `|improved|` too.

File: docs/source/release_notes/release-notes.rst [hunk 5]
```diff
@@ -98,132 +106,157 @@ develop
...
+    :widths: 5 40
+    :header-rows: 1
+
+    *    - Type
+         - Change
+
+    *    - |metrics|
+         - ``SweepMetrics`` are now updated at the end of every batch rather than cumulative metrics at the end of every table.
+           This will provide more accurate metrics for when sweep is doing something.
+           Sweeping run through the sweep endpoint will now also contribute to these metrics — before it didn't update any metrics which again distorted the view of what work sweep was doing on the DB.
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: nit: either _sweeping_ through the sweep endpoint ..., or _Executing a sweep run_ through the sweep endpoint...

File: docs/source/release_notes/release-notes.rst [hunk 6]
```diff
@@ -98,132 +106,157 @@ develop
...
+           provided a value. This defaults to 8 and can be configured with the ``KeyValueServiceConfig#defaultGetRangesConcurrency`` parameter.
+           Check the full configuration docs `here <https://palantir.github.io/atlasdb/html/configuration/key_value_service_configs/index.html>`__.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2484>`__)
+
+    *    - |devbreak|
+         - Simplify and annotate the constructors for ``SerializableTransactionManager``. This should make the code of that class more maintainable.
+           If you used one of the deleted or deprecated constructors, use the static ``create`` method.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2549>`__)
+
+Logs and Metrics
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: See earlier comment on abstraction - I think it's good we have a section on these and having tags makes sense, but these changes should have a core type as well (for consistency with internal structured changelogs). 

I read these as improved, new, fixed, new, improved, improved (and user break), fixed, fixed and fixed respectively.

File: docs/source/release_notes/release-notes.rst [hunk 7]
```diff
@@ -98,132 +106,157 @@ develop
...
+         - Timelock server no longer appends client names to metrics. Instead, each metric is aggregated across all clients.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2501>`__)
+
+    *    - |metrics|
+         - We now report metrics for Transaction conflicts.
+           The metrics are a meter reported under the name ``SerializableTransaction.SerializableTransactionConflict``.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2540>`__)
+
+    *    - |logs|
+         - Specified which logs from Cassandra* classes were Safe or Unsafe for collection, improving the data that we can collect for debugging purposes.
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: nit: either Cassandra or C*

File: docs/source/release_notes/release-notes.rst [hunk 8]
```diff
@@ -98,132 +106,157 @@ develop
...
-          (`Pull Request <https://github.com/palantir/atlasdb/pull/2543>`__)
+    *    - |fixed|
+         - Timelock server will try to gain leadership synchronously when the first time a new client namespace is requested. Previously, the first request would always return 503.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2503>`__)
 
-    *   - |improved|
-        - Specified which logs from Cassandra* classes were Safe or Unsafe for collection, improving the data that we can collect for debugging purposes.
-          (`Pull Request <https://github.com/palantir/atlasdb/pull/2537>`__)
+    *    - |fixed|
+         - ``SerializableErrorDecoder`` will decode error properly instead of throwing ``NullPointerException``.
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: nit: ... will decode _errors_ ...

File: docs/source/release_notes/release-notes.rst [hunk 9]
```diff
@@ -269,13 +302,7 @@ v0.61.0
...
            (`Pull Request <https://github.com/palantir/atlasdb/pull/2409>`__)
 
-    *   - |improved|
-        - Exposes another version of ``getRanges`` that uses a configurable concurrency level when not explicitly
-          provided a value. This defaults to 8 and can be configured with the ``KeyValueServiceConfig#defaultGetRangesConcurrency`` parameter.
-          Check the full configuration docs `here <https://palantir.github.io/atlasdb/html/configuration/key_value_service_configs/index.html>`__.
-          (`Pull Request <https://github.com/palantir/atlasdb/pull/2484>`__)
-
-    *    - |fixed|
+    *    - |logs|
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: assuming earlier model, this should be logs and fixed. Not sure if we need to do retroactive changes though.

File: docs/source/release_notes/release-notes.rst [hunk 10]
```diff
@@ -98,132 +106,157 @@ develop
...
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2446>`__)
 
-    *   - |fixed|
-        - Fixed an issue where a ``waitForLocks`` request could retry unnecessarily.
-          (`Pull Request <https://github.com/palantir/atlasdb/pull/2491>`__)
+    *    - |fixed|
+         - Async Initialization now works with TimeLock Server.
+           Previously, for Cassandra we would attempt to immediately migrate the timestamp bound from Cassandra to TimeLock on startup, which would fail if either of them was unavailable.
+           For DBKVS or other key-value services, we would attempt to ping TimeLock on startup, which would fail if TimeLock was unavailable (though the KVS need not be available).
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2518>`__)
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: Whoops, something's gone wrong here - I think this should be 2520.

File: docs/source/release_notes/release-notes.rst [hunk 11]
```diff
@@ -98,132 +106,157 @@ develop
...
-           current cache size is inadequate.
-           This is a small API change for users manually constructing a TransactionManager, which now requires a
-           transaction cache size parameter. Please add it from the AtlasDbConfig, or instead of manually creating
-           a TransactionManager, utilize the helpers in TransactionManagers to have this done for you.
-           (`Pull Request 1 <https://github.com/palantir/atlasdb/pull/2496>`__)
-           (`Pull Request 2 <https://github.com/palantir/atlasdb/pull/2554>`__)
+    *    - |fixed|
+         - ``AsyncInitializer`` now shuts down its executor after initialization has completed.
+           Previously, the executor service wasn't shut down, which could lead to the initializer thread hanging around unnecessarily.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2518>`__)
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: Yep, I think this should be 2520 as well.

File: docs/source/release_notes/release-notes.rst [hunk 12]
```diff
@@ -98,132 +106,157 @@ develop
...
-           Previously, the executor service wasn't shut down, which could lead to the initializer thread hanging around unnecessarily.
-           (`Pull Request <https://github.com/palantir/atlasdb/pull/2518>`__)
+    *    - |improved|
+         - The duration between attempts of whitelist Cassandra nodes was reduced from 5 minutes to 2 minutes, and the minimum period a node is blacklisted for was reduced from 2 minutes to 30 seconds.
+           This means we check the health of a blacklisted Cassandra node and whitelist it faster than before.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2543>`__)
 
-    *    - |fixed|
+    *    - |improved| |devbreak|
+         - Size of the transaction cache is now configurable. It is not anticipated end users will need to touch this;
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: nit: `The size...`

File: docs/source/release_notes/release-notes.rst [hunk 13]
```diff
@@ -98,132 +106,157 @@ develop
...
+           it is more likely that this will be configured via per-service overrides for the services for whom the
+           current cache size is inadequate.
+           This is a small API change for users manually constructing a TransactionManager, which now requires a
+           transaction cache size parameter. Please add it from the AtlasDbConfig, or instead of manually creating
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: `AtlasDbRuntimeConfig`

File: docs/source/release_notes/release-notes.rst (hunk 12 above)
Comment (gsheasby) [reply docs/source/release_notes/release-notes.rst]: You should note the new parameter name and default value (see the below note), and also that while it's in the `runtime` block, it actually does not yet live reload (link to the relevant issue?)

File: docs/source/release_notes/release-notes.rst [hunk 14]
```diff
@@ -98,132 +106,157 @@ develop
...
+         - The duration between attempts of whitelist Cassandra nodes was reduced from 5 minutes to 2 minutes, and the minimum period a node is blacklisted for was reduced from 2 minutes to 30 seconds.
+           This means we check the health of a blacklisted Cassandra node and whitelist it faster than before.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2543>`__)
 
-    *    - |fixed|
+    *    - |improved| |devbreak|
+         - Size of the transaction cache is now configurable. It is not anticipated end users will need to touch this;
+           it is more likely that this will be configured via per-service overrides for the services for whom the
+           current cache size is inadequate.
+           This is a small API change for users manually constructing a TransactionManager, which now requires a
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: FYI: If you intended a line break here in the actual notes, you'll need an empty line here.

File: docs/source/release_notes/release-notes.rst [hunk 15]
```diff
@@ -98,132 +106,157 @@ develop
...
+Logs and Metrics
+
+.. list-table::
+    :widths: 5 40
+    :header-rows: 1
+
+    *    - Type
+         - Change
+
+    *    - |metrics|
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: Consider also tagging these with (in this case) `|fixed|`.

File: docs/source/release_notes/release-notes.rst (hunk 8 above)
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: error --> errors

File: docs/source/release_notes/release-notes.rst [hunk 16]
```diff
@@ -98,132 +106,157 @@ develop
...
-    *    - |improved| |devbreak|
-         - Size of the transaction cache is now configurable. It is not anticipated end users will need to touch this;
-           it is more likely that this will be configured via per-service overrides for the services for whom the
-           current cache size is inadequate.
-           This is a small API change for users manually constructing a TransactionManager, which now requires a
-           transaction cache size parameter. Please add it from the AtlasDbConfig, or instead of manually creating
-           a TransactionManager, utilize the helpers in TransactionManagers to have this done for you.
-           (`Pull Request 1 <https://github.com/palantir/atlasdb/pull/2496>`__)
-           (`Pull Request 2 <https://github.com/palantir/atlasdb/pull/2554>`__)
+    *    - |fixed|
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: It would be nice to group the async initializer fixes together.

File: docs/source/release_notes/release-notes.rst (hunk 7 above)
Comment (fsamuel-bs) [reply docs/source/release_notes/release-notes.rst]: I meant CassandraKVS, CassandraClientPoolingContainer... Thus "Cassandra* classes"

**Comment by fsamuel-bs** [comment]
I think I've addressed all the comments here. It's ready for review again. Pinging @jeremyk-91 and @gsheasby.
